content/docs/0.16.x/manage/shipyard/index.md

```diff
@@ -1,290 +1,103 @@
+    apiVersion: spec.keptn.sh/0.2.3
+    kind: "Shipyard"
+    metadata:
+      name: "shipyard-sockshop"
+    spec:
+      stages:
+        - name: "dev"
+        - name: "hardening"
+        - name: "production"
 
 **Example:** Extended shipyard with a delivery sequence in all three stage:
 
-```yaml
-apiVersion: spec.keptn.sh/0.2.3
-kind: "Shipyard"
-metadata:
-  name: "shipyard-sockshop"
-spec:
-  stages:
-    - name: "dev"
-      sequences:
-      - name: "delivery"
-        tasks: 
-        - name: "deployment"
-        - name: "release"
-
-    - name: "hardening"
-      sequences:
-      - name: "delivery"
-        triggeredOn:
-          - event: "dev.delivery.finished"
-        tasks: 
-        - name: "deployment"
-        - name: "release"
-
-    - name: "production"
-      sequences:
-      - name: "delivery"
-        triggeredOn:
-          - event: "hardening.delivery.finished"
-        tasks: 
-        - name: "deployment"
-        - name: "release"
-```
-
-### Reserved Keptn Tasks
-
-Keptn supports a set of opinionated tasks for declaring a delivery or remediation sequence:
-
-* action
-* approval
-* deployment
-* evaluation
-* get-action
-* release
-* rollback
-* test
-
-#### Action
-
-The action task indicates that a remediation action should be executed by an action provider. It is used within a [remediation workflow](../../automated_operations/remediation).
-
-**Usage:**
-
-```
-- name: action
-```
-
-#### Approval
-
-The approval task intercepts the task sequence and waits for a user to approve or decline the open approval. This task can be added, for example, before deploying an artifact into the next stage. The approval strategy can be defined based on the evaluation result `pass` and `warning`. Keptn supports the following approval strategies for the evaluation results `pass` and `warning`:
-
-* `automatic`: The artifact is deployed automatically.
-* `manual`: The user is asked for approval before triggering the deployment.
-
-This allows combinations as follows:
-
-|                          | Evaluation result: pass           | Evaluation result: warning                 | Behavior  |
-|--------------------------|-----------------------------------|--------------------------------------------|-----------|
-| **Skip approval task:** | pass:automatic | warning:automatic | Regardless of the evaluation result, the approval task is skipped |
-| **Depending on evaluation result:**   | pass:automatic | warning:manual    | If the evaluation result is a **warning**, an approval is required |
-| **Depending on evaluation result:**   | pass:manual    | warning:automatic | If the evaluation result is a **pass**, an approval is required |
-| **Mandatory approval task:**          | pass:manual    | warning:manual    | Regardless of the evaluation result, an approval is required |
-
-By default, an `automatic` approval strategy is used for evaluation result `pass` and `warning`.
-
-**Usage:**
-
-```
-- name: approval
-  properties: 
```

Comment:
This is 100% correct but FYI: Not all tasks (tools actioning those tasks) require properties so `properties` is optional. In fact, most tools do not use properties. Even those that do typically will have their properties passed in at runtime via JSON data so as to not hardcode it into the shipyard file.

For example:

```
tasks
  - name: evaluation
    properties:
      timeframe: 5m
```
Denotes that we want to run a quality gate evaluation and look at the previous 5 minutes.

This duration is hardcoded into the shipyard file. When keptn generates a cloudevent for the `evaluation.triggered` event, it comes out like this:

```
{
  "id": "1234...",
  "type": "sh.keptn.event.evaluation.triggered",
  # other fields removed for brevity
  "data": {
     "evaluation": {    # notice this is the task name
        "timeframe": "5m"
     }
}
}
```

So you could just as easily pass this in when you trigger the sequence via curl:

```
# JSON payload which triggers the sequence from outside keptn
# Notice the evaluation timeframe is already given by the user
# Meaning the properties block in the shipyard is unneccessary
# Benefit here is that the timeframe is now dynamic
{
  "type": "sh.keptn.event.SomeStage.MySequence.triggered",
  # Other fields removed for brevity
  "data": {
    "evaluation": {
      "timeframe": "5m"
    }
  }
}
```

Reply:
The ref file does say that properties are optional, and I added a note about the JSON

Reply:
I think this is incorporated